consensus/src/chained_bft/block_storage/sync_manager.rs

```diff
@@ -173,6 +174,14 @@ impl<T: Payload> BlockStore<T> {
         // ensure it's [b1, b2]
         blocks.reverse();
         self.rebuild(root, blocks, quorum_certs).await;
+        if highest_ledger_info.ends_epoch() {
+            retriever
+                .network
+                .broadcast_epoch_change(ValidatorChangeEventWithProof::new(vec![
```

Comment:
we pay O(N^2) for this event? Should I read the paper?

Reply:
no, it's not n^2, in happy path the first one who collects the ledger info would broadcast and everyone would call `EpochManager.start_new_epoch` without go into here.

(and we don't need to broadcast here either, I was probably lazy to add a send_to_self api)